src/main/java/com/zufar/icedlatte/review/api/ProductReviewValidator.java

```diff
@@ -57,4 +60,26 @@ public void validateProductReviewDeletionAllowed(final UUID productReviewId) {
         }
     }
 
+    public void validateProductReviewRateUpdateAllowed(final UUID productId, final ProductReviewRateDto request,
+                                                       final ProductReviewRate productReviewRate) {
+        var userId = securityPrincipalProvider.getUserId();
+        var reviewId = request.getProductReviewId();
+        var reviewOwner = productReviewRate.getReview().getUser().getId();
+        if (userId.equals(reviewOwner)) {
+            log.warn("User cannot rate their product review");
+            throw new DeniedProductReviewRateUpdateException(userId, reviewId);
+        }
+        var productIdOfReview = productReviewRate.getReview().getProductInfo().getProductId();
+        if (!productId.equals(productIdOfReview)) {
+            log.warn("Review's product id {} does not correspond to the product id {}",
+                    productIdOfReview, productId);
+            throw new DeniedProductReviewRateUpdateException(userId, reviewId);
+        }
+        var isLike = request.getIsLike();
+        var prevIsLike = productReviewRate.getIsLike();
+        if (prevIsLike == isLike) {
```

Comment:
may be better
if(request.getIsLike().equal(productReviewRate.getIsLike()))
without using intermediate variables